ui/src/app/zones/views/ZonesListTable/ZonesListTable.tsx

```diff
@@ -40,18 +44,26 @@ const ZonesListTable = (): JSX.Element => {
         },
         {
           content: (
-            <Link to={`${machineURLs.machines.index}?zone=%3D${zone.name}`}>
+            <Link to={`${machineURLs.machines.index}${machinesFilter}`}>
```

Comment:
@bethcollins92 @sowasred2012 instead of writing out the query parameters manually, we have this `filtersToQueryString` util that handles all the million ways you can filter machines.